src/generators.rs

```diff
@@ -165,6 +165,34 @@ impl BulletproofGens {
         gens
     }
 
+    /// Generate a set of generators for a single party where some of the
+    /// generators are shared with another proof system.
+    pub fn new_single_party_with_shared_generators(
+        gens_capacity: usize,
+        shared_generators: &[RistrettoPoint],
+        insertion_point: usize,
+        right_side_allocated: bool,
+    ) -> Self {
+        let mut gens = Self::new(gens_capacity, 1);
+
+        let mut index = insertion_point;
+        let mut left_side = !right_side_allocated;
+
+        // Insert the shared generators.
+        for gen in shared_generators.iter() {
```

Comment:
I don't feel *that* strongly about it, but I think the permutation should occur at the layer above Bulletproofs and SDLP. BPs doesn't know how you want the generators to align with inputs, as it doesn't know what your other proof system wants.

Reply:
Option 2.

Reply:
Unless by permutation you meant the reversing of the generators and keeping this function the same.

Reply:
Ok done.